### DISCUSSION TITLE
Automatically publish new documents (disable drafts feature)

### DISCUSSION THREAD

**Discussion by looeee** [outline/outline#3824] [General] [answered]
Is there any way to automatically publish new documents? 

I am migrating a team from Notion to Outline and generally it's going smoothly. This is great software. 

Weirdly though, the biggest blocker for users is getting used to the fact that new documents are not automatically published. 

I would like to disable the draft feature and have new documents show up immediately for all users no matter who creates them. Besides causing confusion it's just not something we need.

**Comment by tommoor** [comment]
There isn’t currently a way but this seems like it would make for a reasonable team preference

**Reply by looeee** [reply]
Should I open a feature request?

**Comment by anthony-crystalpeak** [comment]
I just posed https://github.com/outline/outline/discussions/5298 and am looking for the same feature. Where can we stub out a feature request for this?

**Reply by tommoor** [reply]
Just created an issue linked here

**Comment by blitz** [comment]
> Weirdly though, the biggest blocker for users is getting used to the fact that new documents are not automatically published.

Same here. Even after using Outline for a while, it's easy to get distracted while writing a document and not finding it later, because it's not where it's supposed to be, but in the Drafts folder.

An alternative, would be to have the document in its intended location, just marked as "Draft" instead of hiding somewhere.

**Reply by EduNetEurope** [reply]
Yes, please put the document in its intended location and mark it as a draft.

**Reply by blitz** [reply]
Almost a year later and I still have this issue regularly. :disappointed:

**Reply by EduNetEurope** [reply]
Yes, Outline is great but this is so counterintuitive and against all trends.

**Reply by tommoor** [reply]
We need to have a space for personal documents that is private by default, once this is the case we can default new docs to published wherever you create them, the drafts concept can largely disappear and be replaced by the private docs area… that's the general plan of action here.

**Reply by EduNetEurope** [reply]
Sounds great, @tommoor. I don't think that drafts should disappear, though.
There should be 
- private documents, 
- draft = unpublished, but shared documents that are not finalised yet, and 
- published documents.

This seems to be the norm (we have tested 14 wiki apps).

**Reply by EduNetEurope** [reply]
Is there an ETA for private documents?

**Reply by almereyda** [reply]
@EduNetEurope Can you share your comparison of 14 wiki applications?

**Reply by tommoor** [reply]
We're going to change this behavior for next release so that new documents _**in a collection**_ are published by default. You'll still be able to create drafts from the command menu or the drafts screen.

This will be the behavior for all users and I think it will solve the majority of confusion here with a relatively small change.

**Reply by EduNetEurope** [reply]
> @EduNetEurope Can you share your comparison of 14 wiki applications?

Hi @almereyda, unfortunately there is no systematic list I could share. I just remember the number because I had to put it down in a report four years ago. But, at the time, we looked for very specific features, such as auto-linking of words to glossary entries. MediaWiki and Document360 met the criteria back then. I cannot remember a single one not having the option to select for a document whether it should be marked as a draft or published. I looked at many apps again last year when we needed to find an app to share information internally, and Outline is amazing for us since we can host it (privacy) for free (we are a non-profit). 

If we needed to share collections publicly, though, the upcoming feature announced by @tommoor would not work for us. As I understand it, it would mean that what should be marked as a draft within a publicly shared document / collection would automatically become available online.

**Comment by tommoor** [comment, marked as answer]
This behavior was changed in https://github.com/outline/outline/releases/tag/v0.78.0 – documents created in a collection will be automatically published. You can still create drafts from the command menu or drafts tab.

**Comment by mugilan-sakthivel** [comment]
Hi @tommoor,

Is there a way to disable the automatic publishing feature? Currently, when I create a document, it gets published immediately. If I want to use the webhook to create and publish separately, both actions are happening simultaneously. This causes issues because the document is published before I’ve added any content, and we’re not receiving the correct data.

It would be helpful to have an option to stop the auto-publish behavior so that creation and publishing can be handled independently.

Thanks!

**Comment by EduNetEurope** [comment]
Hi @mugilankani, 

If you create a draft in the draft section, it is not published immediately – but unfortunately, it is not possible to mark a document within a collection as a draft.

We worked with two other apps before, both of which allowed us to:
- mark a document as a draft within a collection
- [hide a document from public view within a publicly shared collection](https://github.com/outline/outline/discussions/9889)

We miss these two features terribly. I am a huge fan of Outline, but the separate draft section does not make sense to me at all.

Here is a related feature request regarding draft revisions of published documents: https://github.com/outline/outline/discussions/6728

**Reply by mugilan-sakthivel** [reply]
Yes, it’s challenging because the platform is evolving without a standardized approach that applies to all users, similar to how a single team might request a change. Currently, we don’t have these features available. Are there any alternative ways to achieve this?

**Reply by almereyda** [reply]
I'm not seeing what you describe with

> it is not possible to mark a document within a collection as a draft

Here are three drafts in two different collections:

<img width="1192" height="704" alt="image" src="https://github.com/user-attachments/assets/07bfcb23-7327-4984-ae33-4aa5b3ce1c9f" />

And here they are listed in the collections:

<img width="1346" height="1114" alt="image" src="https://github.com/user-attachments/assets/baaba3a7-c0e1-430a-84ef-b903d109c386" />

<img width="1122" height="536" alt="image" src="https://github.com/user-attachments/assets/92024eaa-c5bd-43b7-bedc-f609607c320e" />

When you create a document in a collection and turn it a draft, it remains associated to that collection.

**Comment by EduNetEurope** [comment]
The draft is no longer visibly nested within the collection. It will reappear in the collection when it is republished.
In other apps, the draft remains in its place and is somehow marked as a draft (greyed out, in italics, etc.)
